js/Update.elm

```diff
@@ -543,202 +541,153 @@ changeStage stage model =
-                    ++ " during "
-                    ++ toString model
-                )
-            )
-                ( model, Cmd.none )
+type alias EffLens a s =
+    Lens a (Eff a) s (Eff s)
 
 
-tryUpdateGame :
-    (GameModel -> ( GameModel, Cmd Msg ))
-    -> AppModel
-    -> ( AppModel, Cmd Msg )
-tryUpdateGame upd =
-    updateIfGame <|
-        \m ->
-            let
-                ( newM, cmd ) =
-                    upd m
-            in
-                ( Game newM, cmd )
-
-
-updateIfReady :
-    (ReadyModel -> GameModel -> ( GameModel, Cmd Msg ))
-    -> GameModel
-    -> ( GameModel, Cmd Msg )
-updateIfReady upd model =
-    case model.stage of
-        ReadyStage m ->
-            upd m model
-
-        _ ->
-            (Debug.log
-                ("Tried running update function "
-                    ++ toString upd
-                    ++ " during "
-                    ++ toString model.stage
-                )
-                identity
-            )
-                ( model, Cmd.none )
-
-
-tryUpdateReady :
-    (ReadyModel -> ( ReadyModel, Cmd Msg ))
-    -> GameModel
-    -> ( GameModel, Cmd Msg )
-tryUpdateReady upd =
-    updateIfReady <|
-        \m model ->
-            let
-                ( newM, cmd ) =
-                    upd m
-            in
-                ( { model | stage = ReadyStage newM }, cmd )
+goIn : EffLens a b -> EffLens b c -> EffLens a c
+goIn inner outer =
+    { get = outer.get >> Maybe.andThen inner.get
+    , set =
+        \a c ->
+            case outer.get c of
+                Just b ->
+                    outer.set (inner.set a b) c
 
+                Nothing ->
+                    Debug.log "No stage found" c ! []
+    }
 
-tryUpdateProduction :
-    GameModel
-    -> (ProductionModel -> ( ProductionModel, Cmd Msg ))
-    -> ( GameModel, Cmd Msg )
-tryUpdateProduction model upd =
-    case model.stage of
-        ProductionStage m ->
-            let
-                ( newM, cmd ) =
-                    upd m
-            in
-                ( { model | stage = ProductionStage newM }, cmd )
-
-        _ ->
-            (Debug.log
-                ("Tried running update function "
-                    ++ toString upd
-                    ++ " during "
-                    ++ toString model.stage
-                )
-                identity
-            )
-                ( model, Cmd.none )
-
-
-updateIfAuction :
-    (AuctionModel -> GameModel -> ( GameModel, Cmd Msg ))
-    -> GameModel
-    -> ( GameModel, Cmd Msg )
-updateIfAuction upd model =
-    case model.stage of
-        AuctionStage m ->
-            upd m model
-
-        _ ->
-            (Debug.log
-                ("Tried running update function "
-                    ++ toString upd
-                    ++ " during "
-                    ++ toString model.stage
-                )
-            )
-                ( model, Cmd.none )
 
+welcomeL : EffLens WelcomeModel AppModel
+welcomeL =
```

Comment:
Why do we need these L suffix? What is the L supposed to mean?

Reply:
It means "lens". Should we remove it?

Reply:
I think you should remove it, since they all have L. 

The L is clear from the type signature, unless you plan on making a welcome without an L.

Reply:
Okay